### ISSUE TITLE
Server Locks Up / Crashes

### ISSUE THREAD

**Issue by TekPK** [StrangeLoopGames/EcoIssues#4222]
Server running 7.0.7 locked up and crashed. Dump shows this:

```
--BEGIN DUMP--
Dump Time
02/09/2018 21:24:02

Exception
System.Exception: Caught exception invoking RPC PlayerInteract on Player!

Object reference not set to an instance of an object.


Stack:
   at Eco.Core.Controller.ControllerManager.PackageMethodResult(BSONObject bson, INetClient boundClient, IController controller, Tuple`2 method)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Gameplay.Objects.WorldObject.GetBSON(InteractionContext context)
   at Eco.Gameplay.Objects.WorldObject.OnActInteract(InteractionContext context)
   at Eco.Gameplay.Interactions.InteractionExtensions.ExecuteUntilSuccess(IEnumerable`1 results)
   at Eco.Gameplay.Players.Player.PlayerInteract(InteractionInfo info)

   at Eco.Shared.Networking.RPCManager.TryInvoke(Object controller, String methodname, BSONObject bsonArgs, Object& result)
   at Eco.Shared.Networking.RPCManager.InvokeOn(BSONObject bson, Object controller, String methodname)
   at Eco.Shared.Networking.RPCManager.HandleReceiveRPC(INetClient client, BSONObject bson)
   at Eco.Shared.Networking.NetPort.ProcessBuffer()
   at Eco.Shared.Networking.NetPort.Update()
   at Eco.Plugins.Networking.Clients.Client.<.ctor>b__55_0()

--END DUMP--

```

**Comment by TekPK** [comment]
Also getting a lot of these dumps 

--BEGIN DUMP--
Dump Time
02/09/2018 22:45:12

Exception
System.NullReferenceException: Object reference not set to an instance of an object.
   at Eco.Core.Controller.ControllerManager.PackageMethodResult(BSONObject bson, INetClient boundClient, IController controller, Tuple`2 method)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Core.Controller.ControllerManager.PackageValue(BSONObject bson, INetClient boundClient, Object value, Int32 nameID)
   at Eco.Core.Controller.ControllerManager.PackageController(IController controller, INetClient boundClient)
   at Eco.Gameplay.Players.Player.SendInitialState(BSONObject obj, INetObjectViewer viewer)
   at Eco.Shared.Networking.NetObjectManager.GetInitialObjectStates(List`1 newObjects, INetObjectViewer viewer)
   at Eco.Plugins.Networking.Clients.Client.Update()
   at Eco.Plugins.Networking.Clients.Client.<.ctor>b__55_1()

--END DUMP--

**Comment by Senlarus** [comment]
Also this keeps happening to me, console looks like it's having issues checking my login token often resulting in me getting a "timeout" kick from game.
![image](https://user-images.githubusercontent.com/1503963/36058737-7d064d78-0df6-11e8-894b-1cefc6080e33.png)

**Comment by MarjieAVolk** [comment]
I took a look into the code here, and I think what is happening is that a player is trying to perform an action (such as open a crafting station's ui), but before the server can process the action, the player disconnected (or is kicked due to inactivity perhaps... might be related to lots of server lag).

I think we can fix the server crashes with some null checks down in ControllerManager, but improving performance would probably eliminate these too, if I am correct.

**Comment by MarjieAVolk** [comment]
Seems I was at least slightly wrong, because this error message is also coming up as the disconnect error for several people

**Comment by metachronism** [comment]
should be fixed in 7.2
